clippy_lints/src/blocks_in_conditions.rs

```diff
@@ -86,6 +88,20 @@ impl<'tcx> LateLintPass<'tcx> for BlocksInConditions {
                         if expr.span.from_expansion() || ex.span.from_expansion() {
                             return;
                         }
+
+                        // #9911
+                        if for_each_expr_without_closures(block.expr, |e| {
+                            if let ExprKind::Ret(..) = e.kind {
+                                ControlFlow::Break(())
+                            } else {
+                                ControlFlow::Continue(())
+                            }
+                        })
+                        .is_some()
+                        {
+                            return;
+                        }
```

Comment:
This code already exists as `clippy_utils::contains_return()`. You should also update the comment before this `if` if you refactor it to use `contains_return()`.

```suggestion
                        if expr.span.from_expansion() || ex.span.from_expansion() || contains_return(block.expr) {
                            return;
                        }
```

Reply:
Thanks, I rewrote it to use `contains_return`, but I deliberately separated the logic from the original `if` for readability purposes.